🐛[BUG]: README example not working with e2mip://fcn

### Version

main

### On which installation method(s) does this occur?

Pip

### Describe the issue

I get this bug



```
In [3]: >>> import datetime
   ...: >>> from earth2mip.networks import get_model
   ...: >>> from earth2mip.initial_conditions import cds
   ...: >>> from earth2mip.inference_ensemble import run_basic_inference
   ...: >>> time_loop  = get_model("e2mip://fcn", device="cuda:0")
   ...: >>> data_source = cds.DataSource(time_loop.in_channel_names)
   ...: >>> ds = run_basic_inference(time_loop, n=10, data_source=data_source, time=datetime.datetime(2018, 1, 1))
   ...: >>> ds.chunk()

File ~/.local/lib/python3.10/site-packages/modulus/models/afno/afno.py:372, in PatchEmbed.forward(self, x)
    369 def forward(self, x: Tensor) -> Tensor:
    370     B, C, H, W = x.shape
    371     assert (
--> 372         H == self.img_size[0] and W == self.img_size[1]
    373     ), f"Input image size ({H}*{W}) doesn't match model ({self.img_size[0]}*{self.img_size[1]})."
    374     x = self.proj(x).flatten(2).transpose(1, 2)
    375     return x

AssertionError: Input image size (721*1440) doesn't match model (720*1440).
````

this is probably a bug in run_basic_inference.

### Environment details

_No response_

--- a/README.md
+++ b/README.md
@@ -64,7 +64,7 @@ python
 >>> from earth2mip.networks import get_model
 >>> from earth2mip.initial_conditions import cds
 >>> from earth2mip.inference_ensemble import run_basic_inference
->>> time_loop  = get_model("pangu_weather_6", device="cuda:0")
+>>> time_loop  = get_model("e2mip://pangu_6", device="cuda:0")
 >>> data_source = cds.DataSource(time_loop.in_channel_names)
 >>> ds = run_basic_inference(time_loop, n=10, data_source=data_source, time=datetime.datetime(2018, 1, 1))
 >>> ds.chunk()

--- a/earth2mip/inference_ensemble.py
+++ b/earth2mip/inference_ensemble.py
@@ -278,11 +278,13 @@ def run_basic_inference(
 ):
     """Run a basic inference"""
     ds = data_source[time].sel(channel=model.in_channel_names)
+    regridder = regrid.get_regridder(data_source.grid, model.grid).to(model.device)
 
     # TODO make the dtype flexible
     x = torch.from_numpy(ds.values).cuda().type(torch.float)
     # need a batch dimension of length 1
     x = x[None]
+    x = regridder(x)
 
     arrays = []
     times = []
@@ -293,7 +295,7 @@ def run_basic_inference(
             break
 
     stacked = np.stack(arrays)
-    coords = {**ds.coords}
+    coords = dict(lat=model.grid.lat, lon=model.grid.lon)
     coords["channel"] = model.out_channel_names
     coords["time"] = times
     return xarray.DataArray(

--- a/earth2mip/networks/__init__.py
+++ b/earth2mip/networks/__init__.py
@@ -204,6 +204,7 @@ class Inference(torch.nn.Module, time_loop.TimeLoop):
         """The expected size of the second dimension"""
         return self.n_history + 1
 
+    @property
     def device(self) -> torch.device:
         return self.scale.device
 

--- a/earth2mip/regrid.py
+++ b/earth2mip/regrid.py
@@ -59,7 +59,7 @@ class Identity(torch.nn.Module):
         return x
 
 
-class DropLast(torch.nn.Module):
+class DropSouthPole(torch.nn.Module):
     def forward(self, x):
         return x[..., :-1, :]
 
@@ -75,12 +75,11 @@ def _get_tempest_regridder(src: Grid, dest: Grid) -> TempestRegridder:
     return TempestRegridder(map_file.as_posix())
 
 
-
 def get_regridder(src: Grid, dest: Grid):
     if src == dest:
         return Identity()
     elif (src, dest) == (Grid.grid_721x1440, Grid.grid_720x1440):
-        return DropLast()
+        return DropSouthPole()
     else:
         return _get_tempest_regridder(src, dest)
     raise NotImplementedError()

--- a/earth2mip/regrid.py
+++ b/earth2mip/regrid.py
@@ -59,6 +59,11 @@ class Identity(torch.nn.Module):
         return x
 
 
+class DropLast(torch.nn.Module):
+    def forward(self, x):
+        return x[..., :-1, :]
+
+
 def _get_tempest_regridder(src: Grid, dest: Grid) -> TempestRegridder:
     # TODO map data needs to be available for S2S scoring
     config = Settings()
@@ -70,9 +75,12 @@ def _get_tempest_regridder(src: Grid, dest: Grid) -> TempestRegridder:
     return TempestRegridder(map_file.as_posix())
 
 
+
 def get_regridder(src: Grid, dest: Grid):
     if src == dest:
         return Identity()
+    elif (src, dest) == (Grid.grid_721x1440, Grid.grid_720x1440):
+        return DropLast()
     else:
         return _get_tempest_regridder(src, dest)
     raise NotImplementedError()
@@ -89,6 +97,7 @@ def xarray_regrid(src: xarray.Dataset, dest: Grid):
     dest : Grid
         Target grid scheme
     """
+    # TODO refactor this into a regridder
     # Subsample / interpolate lat lon grid
     if np.isin(dest.lat, src.coords["lat"].values).all():
         src = src.sel(lat=dest.lat)

--- a/tests/test_end_to_end.py
+++ b/tests/test_end_to_end.py
@@ -19,7 +19,9 @@ import torch
 import xarray
 import hashlib
 import numpy as np
-from earth2mip.networks import persistence
+from earth2mip.initial_conditions import cds
+from earth2mip.inference_ensemble import run_basic_inference
+from earth2mip.networks import persistence, get_model
 from earth2mip import (
     schema,
     weather_events,
@@ -40,6 +42,7 @@ def checksum_reduce_precision(arr, digits=3):
 
 
 class get_data_source:
+    grid = schema.Grid.grid_721x1440
     def __init__(self, inference):
         arr = xarray.DataArray(
             np.ones([1, len(inference.in_channel_names), 721, 1440]),
@@ -129,3 +132,11 @@ def test_inference_medium_range(tmpdir, regtest):
         metrics.rmse, digits=3
     )
     metrics.info(regtest)
+
+
+@pytest.mark.dlow
+def test_run_basic_inference():
+    time_loop  = get_model("e2mip://fcn", device="cuda:0")
+    data_source = get_data_source(time_loop)
+    ds = run_basic_inference(time_loop, n=10, data_source=data_source, time=datetime.datetime(2018, 1, 1))
+    print(ds)

--- a/tests/test_end_to_end.py
+++ b/tests/test_end_to_end.py
@@ -19,7 +19,6 @@ import torch
 import xarray
 import hashlib
 import numpy as np
-from earth2mip.initial_conditions import cds
 from earth2mip.inference_ensemble import run_basic_inference
 from earth2mip.networks import persistence, get_model
 from earth2mip import (
@@ -43,6 +42,7 @@ def checksum_reduce_precision(arr, digits=3):
 
 class get_data_source:
     grid = schema.Grid.grid_721x1440
+
     def __init__(self, inference):
         arr = xarray.DataArray(
             np.ones([1, len(inference.in_channel_names), 721, 1440]),
@@ -134,9 +134,11 @@ def test_inference_medium_range(tmpdir, regtest):
     metrics.info(regtest)
 
 
-@pytest.mark.dlow
+@pytest.mark.slow
 def test_run_basic_inference():
-    time_loop  = get_model("e2mip://fcn", device="cuda:0")
+    time_loop = get_model("e2mip://fcn", device="cuda:0")
     data_source = get_data_source(time_loop)
-    ds = run_basic_inference(time_loop, n=10, data_source=data_source, time=datetime.datetime(2018, 1, 1))
+    ds = run_basic_inference(
+        time_loop, n=10, data_source=data_source, time=datetime.datetime(2018, 1, 1)
+    )
     print(ds)

